public/app/features/expressions/components/sqlCompletionProvider.ts

```diff
@@ -0,0 +1,42 @@
+import { SelectableValue } from '@grafana/data';
+import { ColumnDefinition, LanguageCompletionProvider, TableDefinition, TableIdentifier } from '@grafana/plugin-ui';
+import { config } from '@grafana/runtime';
+
+import { ALLOWED_FUNCTIONS } from '../utils/metaSqlExpr';
+
+interface CompletionProviderGetterArgs {
+  getFields: (t: TableIdentifier) => Promise<ColumnDefinition[]>;
+  refIds: Array<SelectableValue<string>>;
+}
+
+export const getSqlCompletionProvider: (args: CompletionProviderGetterArgs) => LanguageCompletionProvider =
+  (args) => (monaco, language) => ({
+    ...language,
+    triggerCharacters: [' '],
+    tables: {
+      resolve: async () => {
+        const refIdsToTableDefs = args.refIds.map((refId) => {
+          const tableDef: TableDefinition = {
+            name: refId.label || refId.value || '',
+            completion: refId.label || refId.value || '',
+          };
+          return tableDef;
+        });
+        return refIdsToTableDefs;
+      },
+    },
+    columns: {
+      resolve: async (t?: TableIdentifier) => {
+        if (config.featureToggles.sqlExpressionsColumnAutoComplete) {
+          return await args.getFields({ table: t?.table });
+        } else {
+          return Promise.resolve([]);
```

Comment:
Nit (non-blocking): Pretty sure we can just return an empty array here since the promise has resolved by this point. 

```suggestion
          return [];
```